<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LearnBackground</class>
 <widget class="QDialog" name="LearnBackground">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>826</width>
    <height>545</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Learning Background</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QTextEdit" name="info">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>30</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>40</height>
        </size>
       </property>
       <property name="font">
        <font>
         <pointsize>9</pointsize>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="verticalScrollBarPolicy">
        <enum>Qt::ScrollBarAsNeeded</enum>
       </property>
       <property name="html">
        <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:9pt; font-weight:600; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:5px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Display info text.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="3" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="IplImageWidget" name="srcImg" native="true">
       <property name="minimumSize">
        <size>
         <width>400</width>
         <height>300</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="IplImageWidget" name="foreground" native="true">
       <property name="minimumSize">
        <size>
         <width>400</width>
         <height>300</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="2" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QProgressBar" name="learnBgProgressBar">
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnStart">
       <property name="text">
        <string>Start</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnOk">
       <property name="text">
        <string>Ok</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnCancel">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="settingsBox">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>150</height>
      </size>
     </property>
     <property name="title">
      <string>Settings</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_5">
      <item row="0" column="0">
       <widget class="QGroupBox" name="stgGeneral">
        <property name="title">
         <string>General Settings</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <widget class="QLabel" name="lblNFramesToLearnBG">
           <property name="text">
            <string>Number of frames for learning background</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="spbNFramesToLearnBG">
           <property name="minimum">
            <number>300</number>
           </property>
           <property name="maximum">
            <number>500</number>
           </property>
           <property name="singleStep">
            <number>50</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="lblDelayValue">
           <property name="text">
            <string>Delay before start learning</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="spbDelayValue">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>5</number>
           </property>
           <property name="singleStep">
            <number>1</number>
           </property>
           <property name="value">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QCheckBox" name="cbSimpleBG">
           <property name="text">
            <string>Simple Background</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QGroupBox" name="stgPostLearning">
        <property name="title">
         <string>Smooth Settings</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="0">
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0">
            <widget class="QLabel" name="lblEroshionValue">
             <property name="text">
              <string>Erosion</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QSlider" name="sldEroshionValue">
             <property name="maximum">
              <number>5</number>
             </property>
             <property name="value">
              <number>2</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <widget class="QSpinBox" name="spbErosionValue">
             <property name="maximum">
              <number>5</number>
             </property>
             <property name="value">
              <number>2</number>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="lblDilationValue">
             <property name="text">
              <string>Dilation</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QSlider" name="sldDilationValue">
             <property name="maximum">
              <number>5</number>
             </property>
             <property name="value">
              <number>1</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="1" column="3">
            <widget class="QSpinBox" name="spbDilationValue">
             <property name="maximum">
              <number>5</number>
             </property>
             <property name="value">
              <number>1</number>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="lblSmoothValue">
             <property name="text">
              <string>Smooth Value</string>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QSlider" name="sldSmoothValue">
             <property name="cursor">
              <cursorShape>SizeAllCursor</cursorShape>
             </property>
             <property name="minimum">
              <number>3</number>
             </property>
             <property name="maximum">
              <number>25</number>
             </property>
             <property name="singleStep">
              <number>2</number>
             </property>
             <property name="pageStep">
              <number>2</number>
             </property>
             <property name="value">
              <number>21</number>
             </property>
             <property name="tracking">
              <bool>false</bool>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="2" column="3">
            <widget class="QSpinBox" name="spbSmoothValue">
             <property name="minimum">
              <number>3</number>
             </property>
             <property name="maximum">
              <number>25</number>
             </property>
             <property name="singleStep">
              <number>2</number>
             </property>
             <property name="value">
              <number>21</number>
             </property>
            </widget>
           </item>
           <item row="3" column="0" colspan="2">
            <widget class="QLabel" name="lblThresholdValue">
             <property name="text">
              <string>Threshold Value</string>
             </property>
            </widget>
           </item>
           <item row="3" column="2">
            <widget class="QSlider" name="sldThresholdValue">
             <property name="minimum">
              <number>0</number>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
             <property name="value">
              <number>10</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="3" column="3">
            <widget class="QSpinBox" name="spbThresholdValue">
             <property name="minimum">
              <number>0</number>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
             <property name="value">
              <number>10</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="2">
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>249</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>IplImageWidget</class>
   <extends>QWidget</extends>
   <header>iplimagewidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>btnStart</tabstop>
  <tabstop>btnOk</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>btnCancel</sender>
   <signal>clicked()</signal>
   <receiver>LearnBackground</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>798</x>
     <y>213</y>
    </hint>
    <hint type="destinationlabel">
     <x>796</x>
     <y>31</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sldEroshionValue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spbErosionValue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>458</x>
     <y>98</y>
    </hint>
    <hint type="destinationlabel">
     <x>529</x>
     <y>102</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spbErosionValue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sldEroshionValue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>532</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>457</x>
     <y>98</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sldDilationValue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spbDilationValue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>478</x>
     <y>122</y>
    </hint>
    <hint type="destinationlabel">
     <x>534</x>
     <y>124</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spbDilationValue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sldDilationValue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>517</x>
     <y>118</y>
    </hint>
    <hint type="destinationlabel">
     <x>478</x>
     <y>125</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sldSmoothValue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spbSmoothValue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>424</x>
     <y>142</y>
    </hint>
    <hint type="destinationlabel">
     <x>544</x>
     <y>146</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spbSmoothValue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sldSmoothValue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>530</x>
     <y>148</y>
    </hint>
    <hint type="destinationlabel">
     <x>462</x>
     <y>148</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sldThresholdValue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spbThresholdValue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>465</x>
     <y>165</y>
    </hint>
    <hint type="destinationlabel">
     <x>539</x>
     <y>165</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spbThresholdValue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sldThresholdValue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>521</x>
     <y>171</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>171</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
